test/tstRay.cpp

```diff
@@ -226,6 +323,54 @@ BOOST_AUTO_TEST_CASE(overlap_distance_sphere,
                              unit_sphere) == 0.f);
 }
 
+#define ARBORX_TEST_RAY_SPHERE_INTERSECTION(ray, sphere, t0_ref, t1_ref)       \
+  do                                                                           \
+  {                                                                            \
+    float t0;                                                                  \
+    float t1;                                                                  \
+    BOOST_TEST(ArborX::Experimental::intersection(ray, sphere, t0, t1));       \
+    BOOST_TEST(t0 == t0_ref);                                                  \
+    BOOST_TEST(t1 == t1_ref);                                                  \
+  } while (false)
+
+#define ARBORX_TEST_RAY_SPHERE_NO_INTERSECTION(ray, sphere)                    \
+  do                                                                           \
+  {                                                                            \
+    float t0;                                                                  \
+    float t1;                                                                  \
+    BOOST_TEST(!ArborX::Experimental::intersection(ray, sphere, t0, t1));      \
+    BOOST_TEST((std::isnan(t0) && std::isnan(t1)));                            \
+  } while (false)
+
+BOOST_AUTO_TEST_CASE(ray_sphere_intersection,
+                     *boost::unit_test::tolerance(1e-6f))
+{
+  using ArborX::Sphere;
+  using ArborX::Experimental::Ray;
+
+  constexpr Sphere unit_sphere{{0, 0, 0}, 1};
+
+  auto const sqrtf_3 = std::sqrt(3.f);
+
+  // clang-format off
+  // hit center of the sphere
+  ARBORX_TEST_RAY_SPHERE_INTERSECTION((Ray{{-2, 0, 0}, {1, 0, 0}}), unit_sphere, 1.f, 3.f);
+  ARBORX_TEST_RAY_SPHERE_INTERSECTION((Ray{{0, 3, 0}, {0, -1, 0}}), unit_sphere,2.f, 4.f);
+  ARBORX_TEST_RAY_SPHERE_INTERSECTION((Ray{{0, 0, -4}, {0, 0, 1}}), unit_sphere,3.f, 5.f);
+  ARBORX_TEST_RAY_SPHERE_INTERSECTION((Ray{{-2, -2, -2}, {1, 1, 1}}), unit_sphere, 2 * sqrtf_3 - 1, 2 * sqrtf_3 + 1);
+
+  ARBORX_TEST_RAY_SPHERE_NO_INTERSECTION((Ray{{-2, -2, -2}, {1, 0, 0}}), unit_sphere);
+  //ARBORX_TEST_RAY_SPHERE_NO_INTERSECTION((Ray{{1, 2, 3}, {4, 5, 6}}), unit_sphere);
```

Comment:
Why is this commented out?

Reply:
Because this one will intersect and it will be a pain to compute `tmin` and `tmax`.